app/src/main/java/edu/uc/coffeens/stocktracker/ui/main/MainViewModel.kt

```diff
@@ -28,17 +28,16 @@ class MainViewModel : ViewModel() {
      * Gets the stocks using the stock service.
      */
     fun fetchStock() {
-        stock = stockService.fetchStock()
+        stocks = stockService.fetchStock()
     }
 
     fun save(stocks: Stock) {
-        firestore.collection("Stocks")
-            .document()
-            .set(stocks)
-            .addOnSuccessListener {
+        val document = firestore.collection("Stocks").document()
+        val set = document.set(stocks)
+            set.addOnSuccessListener {
                 Log.d("Firebase", "document saved")
             }
-            .addOnFailureListener {
+            set.addOnFailureListener {
```

Comment:
Added the firestore collection into a document variable, so it is easier to update the local dto, as described in this lecture video : https://www.youtube.com/watch?v=CuP1elpCuEA&t=727s

Reply:
This is fine as it were... it's simply chaining the method onto the set variable above.  Either way gives the same result.